workflowai/core/client/agent.py

```diff
@@ -486,12 +487,22 @@ async def list_models(self) -> list[ModelInfo]:
         Raises:
             ValueError: If the agent has not been registered (schema_id is None).
         """
+
         if not self.schema_id:
             self.schema_id = await self.register()
 
-        response = await self.api.get(
+        data = ListModelsRequest()
+        if self.version and isinstance(self.version, VersionProperties):
```

Comment:
you should add the instructions as an optional arg to the function so we can check for specific cases (as well as requires_tools)

Reply:
https://github.com/WorkflowAI/workflowai-py/pull/66/commits/fdbe09a041b055fac3fe119e1d8bd1412a5feda6